src/libs/elektra/kdb.c

```diff
@@ -1597,52 +1593,52 @@ static bool runGetPhase (KeySet * backends, Key * parentKey, uint16_t phase)
  * Retrieve Keys from the Key database in an atomic and universal way.
  *
  * @pre The @p handle must be a valid KDB handle as returned from kdbOpen().
- * @pre The @p returned KeySet must be a valid KeySet, e.g. constructed
- *     with ksNew().
- * @pre The @p parentKey Key must be a valid Key, e.g. constructed with
- *     keyNew().
+ * @pre The @p returned KeySet must be a valid KeySet, e.g. constructed with ksNew().
+ * @pre The @p returend KeySet must only contain only keys in the `spec:/`,
+ * 	`dir:/`, `user:/`, `system:/`, `default:/` or `proc:/` namespaces.
+ * @pre The @p parentKey Key must be a valid Key, e.g. constructed with keyNew().
+ * @pre The @p parentKey Key must not have read-only name, value or metadata.
+ * @pre The @p parentKey Key must use the `spec:/`, `dir:/`, `user:/`, `system:/`,
+ * 	`default:/`, `proc:/` or cascading namespace.
  *
  * If you pass `NULL` or a key with read-only metadata as @p parentKey,
  * kdbGet() will fail immediately without doing anything.
  * If you pass another invalid @p parentKey, or `NULL` as @p ks or @p handle,
  * kdbGet() will set an error on @p parentKey and then return immediately.
  *
- * **Attention**: If you pass a non-NULL @p parentKey with writable metadata, kdbGet() will **always** remove
- *                any existing errors and warnings from @p parentKey.
+ * @note If you pass a non-NULL @p parentKey with writable metadata, kdbGet() will **always** remove
+ *       any existing errors and warnings from @p parentKey.
  *
- * **Important**: The KDB @p handle DOES NOT store a copy of the retrieved @p ks internally.
- *                This means if you later call kdbSet() with the same @p handle you must make sure to
- *                pass all keys from @p ks, which you do not want to remove.
+ * @warning The KDB @p handle DOES NOT store a copy of the retrieved @p ks internally.
+ *          This means if you later call kdbSet() with the same @p handle you must make sure to
+ *          pass all keys from @p ks, which you do not want to remove.
  *
  * @par Loadable Namespaces
  *
  * Not all namespace can be loaded.
- * `spec:/`, `dir:/`, `user:/` and `system:/` are loadable.
- * `proc:/` keys can be loaded but will never be persisted or cached.
- * `default:/` keys can be returned by kdbGet() but they will always stem from a specification in `spec:/` keys
- * If @p ks contains a key with any other namespace, an error will be returned.
- *
- * If @p parentKey is in any other namespace, an eror will be returned.
  *
- * **Note**: In general it is recommended to use a @p parentKey in the cascading namespace.
+ * - `spec:/`, `dir:/`, `user:/` and `system:/` can be loaded via kdbGet().
+ * - `proc:/` keys can be loaded via kdbGet(), but are not persisted or cached.
+ * - `default:/` keys can be returned by kdbGet() but they will always stem from a specification in `spec:/` keys
```

Comment:
To not make a constraint to `default:` keys added beforehand:

```suggestion
 * - `default:/` keys can be inserted by kdbGet() but they will always stem from a specification in `spec:/` keys
```

Reply:
> To not make a constraint to `default:` keys added beforehand

I accepted the suggestion, but not sure what you wanted to say here. I thought we want `default:/` keys to come from `spec:/` only (see https://github.com/ElektraInitiative/libelektra/pull/4484#discussion_r980332072). Pre-existing `default:/` keys will be removed, if the overlap with a loaded backend. Outside those backends, they will of course be kept, but that's the case for any other namespace too.

Reply:
The change is good then. But it would be good to have these two sentences in the docu, too.